### ISSUE TITLE
Dependency problems

### ISSUE THREAD

**Issue by RafaelKr** [rekalogika/file-src#14]
Currently I'm getting the error:
`Call to undefined method Intervention\\Image\\ImageManager::gd() at /var/www/vhosts/example.com/httpdocs/vendor/rekalogika/file-image/src/ImageResizer.php:69)`

For me the installed version of `intervention/image` is `2.4-dev`. Probably this is related to the following:

Some days ago you fixed the `open_basedir` compatibility in https://github.com/rekalogika/file-src/commit/fd60341633df6d5c22351fe3a03a9424ee8986b3

When I tried to install the fix I got the following error:

```
$ composer req 'rekalogika/file:^1.8.2'
./composer.json has been updated
Running composer update rekalogika/file
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires rekalogika/file ^1.8.2 -> satisfiable by rekalogika/file[1.8.2].
    - rekalogika/file 1.8.2 requires rekalogika/file-contracts ^1.9 -> found rekalogika/file-contracts[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.
  Problem 2
    - rekalogika/file 1.8.2 requires rekalogika/file-contracts ^1.9 -> found rekalogika/file-contracts[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.
    - rekalogika/file-filepond 1.8.0 requires rekalogika/file ^1.8 -> satisfiable by rekalogika/file[1.8.2].
    - rekalogika/file-filepond is locked to version 1.8.0 and an update of this package was not requested.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
```

So I tried to set the minimum-stability to `dev` and probably that's why I got that `intervention/image` which may be causing that initial error.

Could you maybe please check and fix the dependencies of `rekalogika/file` so I can install stable version 1.8.2?

Some more context: Actually I don't want to install `rekalogika/file` itself but want with the following command
`composer req 'rekalogika/file-association:^1.8.2' 'rekalogika/file-association-entity:^1.8.2' 'rekalogika/file-bundle:^1.8.2' 'rekalogika/file-filepond:^1.8.2'` which results in:

```
 composer req 'rekalogika/file-association:^1.8.2' 'rekalogika/file-association-entit
y:^1.8.2' 'rekalogika/file-bundle:^1.8.2' 'rekalogika/file-filepond:^1.8.2' -W
./composer.json has been updated
Running composer update rekalogika/file-association rekalogika/file-association-entity rekalogika/file-bundle rekalogika/file-filepond --with-all-dependencies
Loading composer repositories with package information
Restricting packages listed in "symfony/symfony" to "6.3.*"
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires rekalogika/file-association ^1.8.2 -> satisfiable by rekalogika/file-association[1.8.2].
    - rekalogika/file-association 1.8.2 requires rekalogika/file-association-contracts ^1.9 -> found rekalogika/file-association-contracts[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.
  Problem 2
    - Root composer.json requires rekalogika/file-association-entity ^1.8.2 -> satisfiable by rekalogika/file-association-entity[1.8.2].
    - rekalogika/file-association-entity 1.8.2 requires rekalogika/file-contracts ^1.9 -> found rekalogika/file-contracts[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.
  Problem 3
    - Root composer.json requires rekalogika/file-bundle ^1.8.2 -> satisfiable by rekalogika/file-bundle[1.8.2].
    - rekalogika/file-bundle 1.8.2 requires rekalogika/file ^1.9 -> found rekalogika/file[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.
  Problem 4
    - Root composer.json requires rekalogika/file-filepond ^1.8.2 -> satisfiable by rekalogika/file-filepond[1.8.2].
    - rekalogika/file-filepond 1.8.2 requires rekalogika/file ^1.9 -> found rekalogika/file[dev-main, 1.9.x-dev (alias of dev-main)] but it does not match your minimum-stability.

```

**Comment by RafaelKr** [comment]
I could just confirm that the initial problem is related to the `intervention/image` version. I don't get that 500 by modifying the `Rekalogika\File\Image\ImageResizer` to work with that version:

```diff
- use Intervention\Image\Encoders\AutoEncoder;
- use Intervention\Image\ImageManager;
+ use Intervention\Image\ImageManagerStatic as Image;

[...]

class ImageResizer extends AbstractFileFilter
{
	[...]

    protected function process(): FileInterface
    {
        $ratio = null;

-        $img = ImageManager::gd()
-            ->read($this->getSourceFile()->getContentAsStream()->detach());
+        $img = Image::make($this->getSourceFile()->getContentAsStream()->detach());

        $w = $img->width();
        $h = $img->height();

        if ($this->aspect == self::ASPECTRATIO_SQUARE) {
            if ($w > $h) {
                $img->crop($h, $h);
            } else {
                $img->crop($w, $w);
            }

            $ratio = 1;
        } elseif ($this->aspect == self::ASPECTRATIO_ORIGINAL) {
            $ratio = $w / $h;
        } else {
            throw new \InvalidArgumentException(sprintf(
                'Unknown aspect ratio "%s"',
                $this->aspect
            ));
        }

        $width = $this->maxWidthOrHeight;
        $height = $this->maxWidthOrHeight;

        if ($width / $height > $ratio) {
            $width = $height * $ratio;
        } else {
            $height = $width / $ratio;
        }

        $img->resize((int) round($width), (int) round($height));
-        $encoded = $img->encode(new AutoEncoder);
+        $encoded = $img->encode();

        return $this->getFileRepository()
            ->createFromString(
                $this->getDerivationFilePointer(),
-                $encoded->toString(),
+                (string) $encoded,
            );
    }
}

```

**Comment by priyadi** [comment]
This is caused by f25bbaad40e5006d6d3621cca65777b6941c7417 where I bumped intervention to 3.3, but forgot to sync the version in the subpackage, so tests did not catch it.

I just released 1.8.4 to fix the problem

**Comment by priyadi** [comment]
Thank you for the good words. I'm applying for GitHub Sponsors, but it is yet to be approved by GitHub after 3 days. I think it might take some time.
